R/sim_fixed_family.R

```diff
@@ -50,29 +50,29 @@
 
 
 sim_fixed_family <- function(n, m, q, hsq, k, sib = 0, path = "") {
-  
+
   stopifnot("n needs to be an integer greater than 0" =
-              (n > 0 && class(n) == "numeric" && n == round(n)),
+              (n > 0 && is.numeric(n) && n == round(n)),
             "m needs to be an integer greater than 0" =
-              (m > 0 && class(m) == "numeric" && m == round(m)),
+              (m > 0 && is.numeric(m) && m == round(m)),
             "q needs to be an integer greater than 0 and smaller than m" =
-              (q > 0 && class(q) == "numeric" && q == round(q) && q <= m),
+              (q > 0 && is.numeric(q) && q == round(q) && q <= m),
             "hsq needs to be a number between 0 and 1" =
-              (hsq > 0 && hsq < 1 && class(hsq) == "numeric"),
+              (hsq > 0 && hsq < 1 && is.numeric(hsq)),
             "k needs to be a number between 0 and 1" =
-              (k > 0 && k < 1 && class(k) == "numeric"),
+              (k > 0 && k < 1 && is.numeric(k)),
             "sib needs to be a non-negative integer" =
-              (sib >= 0 && class(sib) == "numeric" && round(sib) == sib),
+              (sib >= 0 && is.numeric(sib) && round(sib) == sib),
```

Comment:
For my understanding, then R would say that is.numeric(c(1, 2, 3)) == TRUE. I do not see this as a smart feature, but in  our checks it would for n, m, q, hsq, k and sib be great if we check that length(x) == 1.